src/python/pants/engine/build_files.py

```diff
@@ -38,6 +41,103 @@ def _key_func(entry):
     return key
 
 
+class LoadedSymbolsToExpose(Collection[str]):
+    def __str__(self):
+        return str([i for i in self])
+
+
+@dataclass(frozen=True)
+class LoadStatement:
+    path: str
+    symbols_to_expose: LoadedSymbolsToExpose
+
+
+class LoadStatements(Collection[LoadStatement]):
+    pass
+
+
+@dataclass(frozen=True)
+class LoadStatementWithContent:
+    path: str
+    content: FileContent
+    symbols_to_expose: LoadedSymbolsToExpose
+
+
+@dataclass(frozen=True)
+class BuildFilesWithLoads:
+    map: Dict[FileContent, Collection[LoadStatementWithContent]]
+
+
+@rule
+async def snapshot_load_statement(statement: LoadStatement) -> LoadStatementWithContent:
+    snapshot = await Get[Snapshot](PathGlobs(globs=(statement.path,)))
+    files_content = await Get[FilesContent](Digest, snapshot.directory_digest)
+    file_content_list = [fc for fc in files_content]
+    if len(file_content_list) != 1:
+        raise ResolveError(f'Tried to load non existing file: "{statement.path}"')
+    file_content = file_content_list[0]
+    return LoadStatementWithContent(
+        path=statement.path, content=file_content, symbols_to_expose=statement.symbols_to_expose
+    )
+
+
+@rule
+async def parse_build_file_for_load_statements(content: FileContent) -> LoadStatements:
+    class LoadParser(ast.NodeVisitor):
+        """A utility class that parses load() calls in BUILD files.
+
+        A load statement is of the form:
+        load("path/to/a:file.py", "symbol1", "symbol2"...)
+
+        This will import "symbol1", "symbol2"... from file "path/to/a:file.py".
```

Comment:
nit: add a note that supporting `:` is to comply with bazel's format.

Reply:
Good catch, thanks!